derive/src/pymodule.rs

```diff
@@ -1,50 +1,73 @@
 use super::Diagnostic;
-use crate::util::{def_to_name, ItemIdent, ItemMeta};
+use crate::util::{def_to_name, meta_into_nesteds, ItemIdent, ItemMeta, ItemType};
 use proc_macro2::{Span, TokenStream as TokenStream2};
 use quote::{quote, quote_spanned, ToTokens};
-use std::collections::HashSet;
-use syn::{parse_quote, spanned::Spanned, Attribute, AttributeArgs, Ident, Item, Meta, NestedMeta};
-
-fn meta_to_vec(meta: Meta) -> Result<Vec<NestedMeta>, Meta> {
-    match meta {
-        Meta::Path(_) => Ok(Vec::new()),
-        Meta::List(list) => Ok(list.nested.into_iter().collect()),
-        Meta::NameValue(_) => Err(meta),
-    }
-}
+use std::collections::HashMap;
+use syn::{parse_quote, spanned::Spanned, AttributeArgs, Ident, Item, Meta, NestedMeta};
 
 #[derive(Default)]
 struct Module {
-    items: HashSet<(ModuleItem, Vec<Meta>)>,
+    items: HashMap<(String, Vec<Meta>), ModuleItem>,
```

Comment:
Do we still have to have the `Vec<Meta>` in the key if we're distinguishing by name now?

Reply:
Yes, we still need it. `Vec<Meta>` is used to allow duplicated names when it has different cfgs. We use it to allow same named function exists one with `#[unix]` and the other one with `#[windows]`